### PR TITLE
GUVNOR-2399: Errors retrieving LockInfo

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.jboss.org/browse/GUVNOR-2399

There was a race condition between checking if the lock file existed and reading it. This has been simplified to just attempt to read it and return the default ```LockInfo``` if the read fails (e.g. it did not exist). The test failed sporadically without the change; but never (at least in my attempts) failed with the change.

### CODE REVIEW

File: guvnor-services/guvnor-services-backend/src/main/java/org/guvnor/common/services/backend/metadata/MetadataCreator.java [hunk 1]
```diff
@@ -150,25 +151,27 @@ private String getCheckinComment() {
...
     }
 
-    private LockInfo retrieveLockInfo(org.uberfire.backend.vfs.Path path) {
-        final org.uberfire.java.nio.file.Path lockPath = Paths.convert(PathFactory.newLock(path));
-        if (!configIOService.exists(lockPath)) {
-            return new LockInfo(false,
-                                "",
-                                path);
-        }
+    private LockInfo retrieveLockInfo( org.uberfire.backend.vfs.Path path ) {
```
Comment (manstis) [inline guvnor-services/guvnor-services-backend/src/main/java/org/guvnor/common/services/backend/metadata/MetadataCreator.java]: Here's the real (non-code formatting) change!

File: guvnor-services/guvnor-services-backend/src/test/java/org/guvnor/common/services/backend/metadata/MetadataCreatorTest.java [hunk 2]
```diff
@@ -29,89 +30,233 @@
...
-        versionRecords.add(createVersionRecord());
+        assertNotNull( metadata );
+        assertNotNull( metadata.getTags() );
+        assertNotNull( metadata.getDiscussion() );
+        assertNotNull( metadata.getVersion() );
+    }
 
-        Metadata metadata = service.create();
+    @Test
+    //See https://issues.jboss.org/browse/GUVNOR-2399
```
Comment (manstis) [inline guvnor-services/guvnor-services-backend/src/test/java/org/guvnor/common/services/backend/metadata/MetadataCreatorTest.java]: New test here!

**Comment by csadilek** [comment]
+1, thank you @manstis !

**Comment by csadilek** [comment]
@psiroky @manstis this prob. should be cherry-picked to 6.3.x and 6.4.x as well. Thanks again.

**Comment by manstis** [comment]
I'll do the PRs on Wednesday.

**Comment by manstis** [comment]
@psiroky @csadilek See:

(6.3.x) https://github.com/droolsjbpm/guvnor/pull/263
(6.4.x) https://github.com/droolsjbpm/guvnor/pull/262

**Pull request closed without merging**